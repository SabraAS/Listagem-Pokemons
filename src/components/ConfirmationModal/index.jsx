import PropTypes from 'prop-types';
import './index.scss';

const ConfirmationModal = ({ items, onClose, onStartNew }) => {
  return (
    <div className="modal-overlay">
      <div className="confirmation-modal">
        <button
          className="confirmation-modal__close-button"
          data-testid="modal-close-button"
          onClick={onClose}
        >
          ×
        </button>

        <div className="confirmation-modal__header">
          <div className="confirmation-modal__check">✓</div>
          <h2 className="confirmation-modal__title">Equipe formada</h2>
          <p className="confirmation-modal__subtitle">
            Sua equipe está pronta!
          </p>
        </div>

        <div className="confirmation-modal__content">
          <div className="confirmation-modal__items">
            {items.map((item) => (
              <div className="confirmation-modal__item" key={item.id}>
                <img
                  alt={item.name}
                  className="confirmation-modal__item-image"
                  src={item.image}
                />
                <h3 className="confirmation-modal__item-name">{item.name}</h3>
                <span className="confirmation-modal__item-info">
                  {item.characteristic}
                </span>
              </div>
            ))}
          </div>

          <div className="confirmation-modal__total">
            <span>Total de pokemons na equipe</span>
            <span>{items.length}</span>
          </div>
        </div>

        <button
          className="confirmation-modal__button"
          data-testid="start-new-team-button"
          onClick={onStartNew}
        >
          Começar nova equipe
        </button>
      </div>
    </div>
  );
};

ConfirmationModal.propTypes = {
  items: PropTypes.array.isRequired,
  onClose: PropTypes.func.isRequired,
  onStartNew: PropTypes.func.isRequired,
};

export default ConfirmationModal;
