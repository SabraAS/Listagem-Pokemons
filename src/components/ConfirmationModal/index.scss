.confirmation-modal {
  align-items: center;
  background-color: rgb(0 0 0 / 20%);
  display: flex;
  height: 100%;
  justify-content: center;
  left: 0;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 999;

  &__content {
    align-items: center;
    background-color: #fff;
    border-radius: 8px;
    box-shadow: 0 4px 8px rgb(0 0 0 / 10%);
    display: flex;
    flex-direction: column;
    height: 400px;
    justify-content: space-between;
    padding: 24px;
    width: 500px;
  }

  &__close-button {
    align-self: flex-end;
    background: none;
    border: none;
    color: #4b5563;
    cursor: pointer;
    font-size: 24px;
    height: 32px;
    padding: 0;
    width: 32px;
  }

  &__header {
    align-items: center;
    display: flex;
    flex-direction: column;
  }

  &__check {
    align-items: center;
    background-color: #f4dc25;
    border-radius: 100%;
    color: #fff;
    display: flex;
    font-size: 24px;
    height: 48px;
    justify-content: center;
    margin-bottom: 16px;
    width: 48px;
  }

  &__title {
    color: #000;
    font-size: 24px;
    font-weight: 600;
    margin-bottom: 8px;
  }

  &__subtitle {
    color: #4b5563;
    font-size: 16px;
    margin-bottom: 16px;
  }

  &__list {
    display: flex;
    flex-direction: column;
    height: 100%;
    overflow-y: auto;
    width: 100%;
  }

  &__item {
    align-items: center;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    padding: 8px 0;
    width: 100%;
  }

  &__info {
    align-items: center;
    display: flex;
    flex-direction: row;
    gap: 8px;
    justify-content: flex-start;
  }

  &__image {
    height: 32px;
    width: 32px;
  }

  &__name {
    color: #000;
    font-size: 16px;
    font-weight: 600;
  }

  &__characteristic {
    color: #4b5563;
    font-size: 14px;
  }

  &__footer {
    margin-top: auto;
    width: 100%;
  }

  &__total {
    align-items: center;
    border-top: 1px solid #e5e7eb;
    color: #000;
    display: flex;
    flex-direction: row;
    font-size: 16px;
    font-weight: 600;
    justify-content: space-between;
    margin-bottom: 16px;
    padding: 16px 0;
  }

  &__button {
    background-color: #f4dc25;
    border: none;
    border-radius: 8px;
    color: #374151;
    cursor: pointer;
    font-size: 16px;
    font-weight: 600;
    padding: 12px 0;
    width: 100%;
  }
}
