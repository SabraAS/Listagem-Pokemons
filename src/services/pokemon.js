import axios from 'axios';

const BASE_URL = 'https://pokeapi.co/api/v2';

export const getPokemonList = async (limit = 40) => {
  try {
    const response = await axios.get(`${BASE_URL}/pokemon?limit=${limit}`);
    const pokemonList = response.data.results;

    return await Promise.all(
      pokemonList.map(async (pokemon) => {
        const pokemonData = await axios.get(pokemon.url);
        const description = await getPokemonCharacteristic(pokemonData.data.id);
        return {
          id: pokemonData.data.id,
          name: pokemonData.data.name,
          image: pokemonData.data.sprites?.other?.dream_world?.front_default,
          abilities: pokemonData.data.abilities,
          types: pokemonData.data.types,
          characteristic: description,
        };
      }),
    );
  } catch (error) {
    console.log('Erro ao buscar lista de Pokémon');
    throw error;
  }
};

export const getPokemonCharacteristic = async (id) => {
  try {
    const response = await axios.get(`${BASE_URL}/characteristic/${id}`);
    const data = response.data;
    const englishDescription = data.descriptions.find(
      (desc) => desc.language.name === 'en',
    );
    return englishDescription?.description || 'não possui';
  } catch (_e) {
    console.log('Erro ao buscar característica do Pokémon');
    return 'não possui';
  }
};
