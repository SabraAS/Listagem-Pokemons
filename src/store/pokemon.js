import { create } from 'zustand';

export const usePokemonStore = create((set) => ({
  pokemons: [],
  setPokemons: (pokemons) => set({ pokemons }),
  addPokemon: (pokemon) =>
    set((state) => ({ pokemons: [...state.pokemons, pokemon] })),
  removePokemon: (pokemonId) =>
    set((state) => ({
      pokemons: state.pokemons.filter((pokemon) => pokemon.id !== pokemonId),
    })),
  clearTeam: () => set({ pokemons: [] }),
}));
