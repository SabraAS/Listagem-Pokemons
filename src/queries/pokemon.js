import { useQuery } from '@tanstack/react-query';

import { getPokemonList } from '@/services/pokemon';

export const POKEMON_KEYS = {
  all: ['pokemons'],
  list: (limit) => [...POKEMON_KEYS.all, 'list', limit],
};

export const usePokemons = (limit = 20) => {
  return useQuery({
    queryKey: POKEMON_KEYS.list(limit),
    queryFn: () => getPokemonList(limit),
    retry: false,
    refetchOnWindowFocus: false,
    select: (response) => response,
  });
};
