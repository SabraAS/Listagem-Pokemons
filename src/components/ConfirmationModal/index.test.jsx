import { fireEvent, render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import { axe } from 'vitest-axe';

import ConfirmationModal from './index';

import { mockPokemons, mockSinglePokemon } from '@/test/mocks/pokemon';

describe('ConfirmationModal', () => {
  const defaultProps = {
    pokemons: mockPokemons,
    onClose: vi.fn(),
    onStartNewTeam: vi.fn(),
  };

  const singlePokemonProps = {
    pokemons: mockSinglePokemon,
    onClose: vi.fn(),
    onStartNewTeam: vi.fn(),
  };

  describe('Rendering', () => {
    it('should match snapshot', () => {
      const { container } = render(<ConfirmationModal {...defaultProps} />);
      expect(container).toMatchSnapshot();
    });

    it('should render modal title correctly', () => {
      render(<ConfirmationModal {...defaultProps} />);
      expect(
        screen.getByTestId('confirmation-modal-title'),
      ).toBeInTheDocument();
      expect(screen.getByTestId('confirmation-modal-title')).toHaveClass(
        'confirmation-modal__title',
      );
    });

    it('should render subtitle correctly', () => {
      render(<ConfirmationModal {...defaultProps} />);
      expect(
        screen.getByTestId('confirmation-modal-subtitle'),
      ).toBeInTheDocument();
      expect(screen.getByTestId('confirmation-modal-subtitle')).toHaveClass(
        'confirmation-modal__subtitle',
      );
    });

    it('should render check icon', () => {
      render(<ConfirmationModal {...defaultProps} />);
      const check = screen.getByTestId('confirmation-modal-check');
      expect(check).toBeInTheDocument();
      expect(check).toHaveClass('confirmation-modal__check');
    });

    it('should render basic modal structure', () => {
      render(<ConfirmationModal {...defaultProps} />);
      const title = screen.getByTestId('confirmation-modal-title');
      const subtitle = screen.getByTestId('confirmation-modal-subtitle');
      expect(title).toBeInTheDocument();
      expect(title).toHaveClass('confirmation-modal__title');
      expect(subtitle).toBeInTheDocument();
      expect(subtitle).toHaveClass('confirmation-modal__subtitle');
    });

    it('should render pokemon image with correct alt text', () => {
      render(<ConfirmationModal {...defaultProps} />);
      const bulbasaurImg = screen.getByTestId(
        'confirmation-modal-pokemon-image-1',
      );
      expect(bulbasaurImg).toHaveAttribute('alt', 'Imagem do bulbasaur');
    });

    it('should render pokemon characteristics', () => {
      render(<ConfirmationModal {...defaultProps} />);
      const characteristic1 = screen.getByTestId(
        'confirmation-modal-pokemon-characteristic-1',
      );
      expect(characteristic1).toBeInTheDocument();
      expect(characteristic1).toHaveTextContent(
        'A strange seed was planted on its back at birth. The plant sprouts and grows with this Pokémon.',
      );
      const characteristic2 = screen.getByTestId(
        'confirmation-modal-pokemon-characteristic-4',
      ); // id do charmander é 4
      expect(characteristic2).toBeInTheDocument();
      expect(characteristic2).toHaveTextContent(
        'Obviously prefers hot places. When it rains, steam is said to spout from the tip of its tail.',
      );
    });

    it('should render pokemon names with correct classes', () => {
      render(<ConfirmationModal {...defaultProps} />);
      const name1 = screen.getByTestId('confirmation-modal-pokemon-name-1');
      const name2 = screen.getByTestId('confirmation-modal-pokemon-name-4'); // id do charmander é 4
      expect(name1).toHaveClass('confirmation-modal__name');
      expect(name2).toHaveClass('confirmation-modal__name');
    });

    it('should show correct pokemon total', () => {
      render(<ConfirmationModal {...defaultProps} />);
      const total = screen.getByTestId('confirmation-modal-total');
      expect(total).toBeInTheDocument();
      expect(total).toHaveTextContent('2');
    });

    it('should show total 1 when there is only one pokemon', () => {
      render(<ConfirmationModal pokemons={[mockSinglePokemon]} />);
      const total = screen.getByTestId('confirmation-modal-total');
      expect(total).toBeInTheDocument();
      expect(total).toHaveTextContent('1');
    });

    it('should render close button with ×', () => {
      render(<ConfirmationModal {...defaultProps} />);
      const closeButton = screen.getByTestId('confirmation-modal-close-button');
      expect(closeButton).toBeInTheDocument();
      expect(closeButton).toHaveTextContent('×');
    });

    it('should render "Começar nova equipe" button', () => {
      render(<ConfirmationModal {...defaultProps} />);
      const newTeamButton = screen.getByText('Começar nova equipe');
      expect(newTeamButton).toBeInTheDocument();
    });
  });

  describe('Interactions', () => {
    it('should call onClose when × button is clicked', () => {
      const mockOnClose = vi.fn();
      render(<ConfirmationModal {...defaultProps} onClose={mockOnClose} />);
      const closeButton = screen.getByTestId('confirmation-modal-close-button');
      fireEvent.click(closeButton);
      expect(mockOnClose).toHaveBeenCalledTimes(1);
    });

    it('should call onStartNewTeam when "Começar nova equipe" button is clicked', () => {
      const mockOnStartNewTeam = vi.fn();
      render(
        <ConfirmationModal
          {...defaultProps}
          onStartNewTeam={mockOnStartNewTeam}
        />,
      );
      const newTeamButton = screen.getByText('Começar nova equipe');
      fireEvent.click(newTeamButton);
      expect(mockOnStartNewTeam).toHaveBeenCalledTimes(1);
    });

    it('should allow multiple clicks on buttons', () => {
      const mockOnClose = vi.fn();
      const mockOnStartNewTeam = vi.fn();
      render(
        <ConfirmationModal
          {...defaultProps}
          onClose={mockOnClose}
          onStartNewTeam={mockOnStartNewTeam}
        />,
      );
      const closeButton = screen.getByTestId('confirmation-modal-close-button');
      const newTeamButton = screen.getByTestId('confirmation-modal-button');
      fireEvent.click(closeButton);
      fireEvent.click(newTeamButton);
      fireEvent.click(closeButton);
      expect(mockOnClose).toHaveBeenCalledTimes(2);
      expect(mockOnStartNewTeam).toHaveBeenCalledTimes(1);
    });
  });

  describe('Edge cases', () => {
    it('should handle pokemons with incomplete data', () => {
      const incompletePokemons = [
        { id: 1, name: '', image: '', characteristic: '' },
        { id: 2, name: 'test', image: 'test.jpg', characteristic: 'test char' },
      ];
      const incompleteProps = {
        ...defaultProps,
        pokemons: incompletePokemons,
      };
      render(<ConfirmationModal {...incompleteProps} />);
      const name1 = screen.getByTestId('confirmation-modal-pokemon-name-1');
      const name2 = screen.getByTestId('confirmation-modal-pokemon-name-2');
      const characteristic1 = screen.getByTestId(
        'confirmation-modal-pokemon-characteristic-1',
      );
      const characteristic2 = screen.getByTestId(
        'confirmation-modal-pokemon-characteristic-2',
      );
      expect(name1).toHaveTextContent('pokémon sem nome');
      expect(name2).toHaveTextContent('test');
      expect(characteristic1).toHaveTextContent('não possui característica');
      expect(characteristic2).toHaveTextContent('test char');
      const total = screen.getByTestId('confirmation-modal-total');
      expect(total).toBeInTheDocument();
      expect(total).toHaveTextContent('2');
    });

    it('should handle many pokemons', () => {
      const manyPokemons = Array.from({ length: 10 }, (_, i) => ({
        id: i + 1,
        name: `pokemon${i + 1}`,
        image: `image${i + 1}.jpg`,
        characteristic: `char${i + 1}`,
      }));
      const manyPokemonsProps = {
        ...defaultProps,
        pokemons: manyPokemons,
      };
      render(<ConfirmationModal {...manyPokemonsProps} />);
      const total = screen.getByTestId('confirmation-modal-total');
      expect(total).toBeInTheDocument();
      expect(total).toHaveTextContent('10');
    });

    it('should handle pokemon with null ID', () => {
      const pokemonWithNullId = {
        ...mockSinglePokemon,
        id: null,
        name: 'pokemonSemId',
      };
      render(
        <ConfirmationModal
          onClose={vi.fn()}
          onStartNewTeam={vi.fn()}
          pokemons={[pokemonWithNullId]}
        />,
      );

      expect(screen.getByText('pokemonSemId')).toBeInTheDocument();
    });

    it('should handle pokemon with empty name', () => {
      const pokemonWithEmptyName = {
        ...mockSinglePokemon,
        name: '',
      };
      render(
        <ConfirmationModal
          onClose={vi.fn()}
          onStartNewTeam={vi.fn()}
          pokemons={[pokemonWithEmptyName]}
        />,
      );

      expect(screen.getByText('pokémon sem nome')).toBeInTheDocument();

      const imgAlt = screen.getByAltText('Imagem do pokémon');
      expect(imgAlt).toBeInTheDocument();
    });

    it('should handle pokemon with empty characteristic', () => {
      const pokemonWithEmptyChar = {
        ...mockSinglePokemon,
        characteristic: '',
      };
      render(
        <ConfirmationModal
          onClose={vi.fn()}
          onStartNewTeam={vi.fn()}
          pokemons={[pokemonWithEmptyChar]}
        />,
      );

      const characteristic = screen.getByTestId(
        'confirmation-modal-pokemon-characteristic-1',
      );
      expect(characteristic).toHaveTextContent('não possui característica');
    });

    it('should handle pokemon with null image', () => {
      const pokemonWithNullImage = {
        ...mockSinglePokemon,
        image: null,
      };
      render(
        <ConfirmationModal
          onClose={vi.fn()}
          onStartNewTeam={vi.fn()}
          pokemons={[pokemonWithNullImage]}
        />,
      );

      const image = screen.getByAltText('Imagem do bulbasaur');
      expect(image.getAttribute('src')).toBe('');
    });
  });

  describe('CSS structure', () => {
    it('should apply correct CSS classes to main elements', () => {
      render(<ConfirmationModal {...defaultProps} />);
      const modal = screen
        .getByText('Equipe formada')
        .closest('.confirmation-modal');
      expect(modal).toHaveClass('confirmation-modal');
      expect(modal).toHaveAttribute('role', 'dialog');
      expect(modal).toHaveAttribute('aria-modal', 'true');

      const content = screen
        .getByText('Equipe formada')
        .closest('.confirmation-modal__content');
      expect(content).toHaveClass('confirmation-modal__content');

      const check = screen.getByTestId('confirmation-modal-check');
      expect(check).toHaveClass('confirmation-modal__check');
    });

    it('should have correct structure for each list item', () => {
      render(<ConfirmationModal {...defaultProps} />);
      const bulbasaurItem = screen
        .getByText('bulbasaur')
        .closest('.confirmation-modal__item');
      expect(bulbasaurItem).toBeInTheDocument();
      expect(bulbasaurItem).toHaveClass('confirmation-modal__item');

      const info = screen
        .getByText('bulbasaur')
        .closest('.confirmation-modal__info');
      expect(info).toBeInTheDocument();
      expect(info).toHaveClass('confirmation-modal__info');

      const name = screen.getByText('bulbasaur');
      expect(name).toHaveClass('confirmation-modal__name');

      const characteristic = screen.getByText(
        'A strange seed was planted on its back at birth. The plant sprouts and grows with this Pokémon.',
      );
      expect(characteristic).toHaveClass('confirmation-modal__characteristic');

      const image = screen.getByAltText('Imagem do bulbasaur');
      expect(image).toHaveClass('confirmation-modal__image');
    });

    it('should apply correct classes to header', () => {
      render(<ConfirmationModal {...defaultProps} />);
      const header = screen
        .getByText('Equipe formada')
        .closest('.confirmation-modal__header');
      expect(header).toBeInTheDocument();
      expect(header).toHaveClass('confirmation-modal__header');

      const title = screen.getByText('Equipe formada');
      expect(title).toHaveClass('confirmation-modal__title');

      const subtitle = screen.getByText('Sua equipe está pronta!');
      expect(subtitle).toHaveClass('confirmation-modal__subtitle');
    });

    it('should apply correct classes to footer', () => {
      render(<ConfirmationModal {...defaultProps} />);
      const footer = screen
        .getByText('Total de pokémons na equipe:')
        .closest('.confirmation-modal__footer');
      expect(footer).toBeInTheDocument();
      expect(footer).toHaveClass('confirmation-modal__footer');

      const total = screen
        .getByText('Total de pokémons na equipe:')
        .closest('.confirmation-modal__total');
      expect(total).toHaveClass('confirmation-modal__total');

      const button = screen.getByText('Começar nova equipe');
      expect(button).toHaveClass('confirmation-modal__button');
    });

    it('should have correct close button class', () => {
      render(<ConfirmationModal {...defaultProps} />);
      const closeButton = screen.getByTestId('confirmation-modal-close-button');
      expect(closeButton).toHaveClass('confirmation-modal__close-button');
    });

    it('should have correct list container class', () => {
      render(<ConfirmationModal {...defaultProps} />);
      const list = screen
        .getByText('bulbasaur')
        .closest('.confirmation-modal__list');
      expect(list).toBeInTheDocument();
      expect(list).toHaveClass('confirmation-modal__list');
    });
  });

  describe('Accessibility', () => {
    it('should have no accessibility violations', async () => {
      const { container } = render(<ConfirmationModal {...defaultProps} />);
      const results = await axe(container);
      expect(results).toHaveNoViolations();
    });

    it('should have no accessibility violations with empty team', async () => {
      const emptyProps = { ...defaultProps, pokemons: [] };
      const { container } = render(<ConfirmationModal {...emptyProps} />);
      const results = await axe(container);
      expect(results).toHaveNoViolations();
    });

    it('should have proper button accessibility', () => {
      render(<ConfirmationModal {...defaultProps} />);
      const closeButton = screen.getByTestId('confirmation-modal-close-button');
      const newTeamButton = screen.getByText('Começar nova equipe');
      expect(closeButton.tagName).toBe('BUTTON');
      expect(newTeamButton.tagName).toBe('BUTTON');
      expect(closeButton).not.toHaveAttribute('tabindex', '-1');
      expect(newTeamButton).not.toHaveAttribute('tabindex', '-1');
      expect(closeButton).toHaveAttribute('aria-label', 'Fechar modal');
    });

    it('should have proper image accessibility', () => {
      render(<ConfirmationModal {...defaultProps} />);
      const images = screen.getAllByRole('img');
      images.forEach((img) => {
        expect(img).toHaveAttribute('alt');
        expect(img.getAttribute('alt')).not.toBe('');
      });
    });

    it('should have descriptive content for screen readers', () => {
      render(<ConfirmationModal {...defaultProps} />);
      expect(screen.getByText('Equipe formada')).toBeInTheDocument();
      expect(screen.getByText('Sua equipe está pronta!')).toBeInTheDocument();
      expect(
        screen.getByText('Total de pokémons na equipe:'),
      ).toBeInTheDocument();
      expect(screen.getByText('Começar nova equipe')).toBeInTheDocument();
    });

    it('should handle keyboard navigation properly', () => {
      render(<ConfirmationModal {...defaultProps} />);
      const buttons = screen.getAllByRole('button');
      buttons.forEach((button) => {
        expect(button).not.toHaveAttribute('tabindex', '-1');

        button.focus();
        expect(document.activeElement).toBe(button);
      });
    });
  });
});
