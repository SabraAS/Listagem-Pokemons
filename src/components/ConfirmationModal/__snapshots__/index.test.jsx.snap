// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ConfirmationModal > should match snapshot 1`] = `
<div>
  <div
    aria-labelledby="modal-title"
    aria-modal="true"
    class="confirmation-modal"
    role="dialog"
  >
    <div
      class="confirmation-modal__content"
    >
      <button
        aria-label="Fechar modal"
        class="confirmation-modal__close-button"
        data-testid="modal-close-button"
      >
        ×
      </button>
      <div
        class="confirmation-modal__header"
      >
        <div
          class="confirmation-modal__check"
        >
          ✓
        </div>
        <h2
          class="confirmation-modal__title"
          id="modal-title"
        >
          Equipe formada
        </h2>
        <p
          class="confirmation-modal__subtitle"
        >
          Sua equipe está pronta!
        </p>
      </div>
      <div
        class="confirmation-modal__list"
      >
        <div
          class="confirmation-modal__item"
        >
          <div
            class="confirmation-modal__info"
          >
            <img
              alt="Imagem do bulbasaur"
              class="confirmation-modal__image"
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg"
            />
            <p
              class="confirmation-modal__name"
            >
              bulbasaur
            </p>
          </div>
          <p
            class="confirmation-modal__characteristic"
          >
            Loves to eat
          </p>
        </div>
        <div
          class="confirmation-modal__item"
        >
          <div
            class="confirmation-modal__info"
          >
            <img
              alt="Imagem do charmander"
              class="confirmation-modal__image"
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/4.svg"
            />
            <p
              class="confirmation-modal__name"
            >
              charmander
            </p>
          </div>
          <p
            class="confirmation-modal__characteristic"
          >
            Highly curious
          </p>
        </div>
      </div>
      <div
        class="confirmation-modal__footer"
      >
        <div
          class="confirmation-modal__total"
        >
          <p>
            Total de pokémons na equipe:
          </p>
          <p>
            2
          </p>
        </div>
        <button
          class="confirmation-modal__button"
        >
          Começar nova equipe
        </button>
      </div>
    </div>
  </div>
</div>
`;
